src/box/iproto.cc

```diff
@@ -1797,12 +1966,31 @@ tx_process_call_on_yield(struct trigger *trigger, void *event)
 	return 0;
 }
 
+/**
+ * Because user can use iproto call request or iproto sql request to begin,
+ * commit, or rollback transaction, we must update all transaction related
+ * members of the stream structure after processing this type of requests.
+ */
+static void
+tx_set_txn_after(struct iproto_stream *stream, struct txn *prev_txn)
+{
+	if (stream != NULL) {
+		stream->txn = in_txn();
+		if (prev_txn == NULL && stream->txn != NULL)
```

Comment:
&& stream->txn != prev_txn ?

Reply:
I reworked this part of code, now it's simpler and more clear.